### PR TITLE
App Service - generate Azure App Service extension config files

### PR DESCRIPTION
**Pull request by jasmingoh**
This PR adds 2 files (`.deployment` and `.vscode/settings.json`) to the generated project. This enhances the experience of using the Azure App Service VS Code extension (i.e. they only have to run the deploy command and don't have to go through every prompts).

Changes included in this PR:
Since we need the id of the web app generated,
- the createWebApp method returns the deployment id of the object created
- the deployWebApp returns the App Service id (required by the Azure App Service extension)

The id is then passed into the method that generates the `settings.json` file (this tells the Azure App Service extension to deploy the generated project to this web app)
As for the .deployment file, it is required so that the Azure App Service extension will run a build command after the generated project is deployed to the container

To test:
- [x] Launch the extension in preview mode
- [x] Take note of the project name (you'll need it to find the resource group created)
- [x] Please choose React or Vue ** as frontend, Node as backend (Flask is still in progress)
- [x] Add App Service on the Azure page (do "Create a resource group")
- [x] Create the project
- [x] Open the project
- [x] When it finishes deployment, check if there is a `.deployment` file in the root, and a `settings.json` file within the `.vscode` folder
- [x] Do a Ctrl + Shift + P, type in `App Service: Deploy to Web App...` and hit enter
- [x] Click on "Browse website" once the deployment is done

### **Please clean up all the resources created for testing**
** Angular is handled in another PR

### CODE REVIEW

**Comment by jasmingoh** [comment]
Closing it for now because we found out what's required to run the build command on the Azure App Service extension

File: src/extension/src/azure/azure-app-service/appServiceProvider.ts [hunk 1]
```diff
@@ -46,10 +46,11 @@ const APP_SERVICE_DEPLOYMENT_SUFFIX = "-app-service";
 export class AppServiceProvider {
   private webClient: WebSiteManagementClient | undefined;
 
+  // returns the id of the instance created
   public async createWebApp(
     selections: AppServiceSelections,
     appPath: string
-  ): Promise<void> {
+  ): Promise<string | undefined> {
```
Comment (streetcoder123) [inline src/extension/src/azure/azure-app-service/appServiceProvider.ts:53]: Why is it string or undefined?

**Review by streetcoder123 — APPROVED** [review]
Awesome! Deployed my App :)

File: src/extension/src/azure/azure-app-service/appServiceProvider.ts (hunk 1 above)
Comment (jasmingoh) [reply src/extension/src/azure/azure-app-service/appServiceProvider.ts:53]: the result.id is of type string or undefined

**Review by jcoc611-microsoft — APPROVED** [review]

**Review by jsondoo — COMMENTED** [review]
deployed my Vue app here https://myapp156-201989132141.azurewebsites.net/Blank
nice :)

**Review by jsondoo — APPROVED** [review]

**Review by Tanya0609 — APPROVED** [review]

**Pull request merged**